frontend/src/employee-frontend/components/placement-draft/PlacementDraft.tsx

```diff
@@ -197,6 +207,66 @@ export default React.memo(function PlacementDraft() {
     )
   }, [applicationId, redirectToMainPage])
 
+  const isPreschoolPlacement = (placementType: PlacementType) =>
+    placementType === 'PRESCHOOL' ||
+    placementType === 'PREPARATORY' ||
+    placementType === 'PRESCHOOL_DAYCARE' ||
+    placementType === 'PRESCHOOL_DAYCARE_ONLY' ||
+    placementType === 'PREPARATORY_DAYCARE'
+
+  useEffect(() => {
```

Comment:
useEffectin käyttöä tulisi välttää aina kun se on mahdollista. Sen sijaan `preschoolDatesAreValid` voi määritellä näin:
```ts
const preschoolDatesAreValid = useMemo(() => {
  if (
    preschoolTermsResult.isSuccess &&
    placementDraft.isSuccess &&
    isPreschoolPlacement(placementDraft.value.type) &&
    formState.period !== null
  ) {
...
    return preschoolPeriodIsValid && preschoolDaycarePeriodIsValid
  }
  return true
}, [preschoolTermsResult, placementDraft, formState])
```
Ja poistaa ylempää tämän:
```ts
const [preschoolDatesAreValid, setPreschoolDatesAreValid] = useState(true)
```